src/sinks/new_relic/model.rs

```diff
@@ -57,39 +41,88 @@ impl TryFrom<Vec<Event>> for MetricsApiModel {
-                        metric_array.push((
-                            Value::from(metric.name().to_owned()),
+
+                    // Set name, value, and timestamp
+                    metric_data.insert("name".to_owned(), Value::from(series.name.name));
+                    metric_data.insert(
+                        "value".to_owned(),
+                        Value::from(
+                            NotNan::new(value)
+                                .map_err(|_| NewRelicSinkError::new("NaN value not supported"))?,
+                        ),
+                    );
+                    metric_data.insert(
+                        "timestamp".to_owned(),
+                        Value::from(
+                            data.time
+                                .timestamp
+                                .unwrap_or_else(|| DateTime::<Utc>::from(SystemTime::now()))
+                                .timestamp(),
+                        ),
+                    );
+
+                    if let Some(tags) = series.tags {
+                        metric_data.insert(
+                            "attributes".to_owned(),
                             Value::from(
-                                NotNan::new(*value).map_err(|_| {
-                                    NewRelicSinkError::new("NaN value not supported")
-                                })?,
+                                tags.iter_single()
+                                    .map(|(key, value)| (key.to_string(), Value::from(value)))
+                                    .collect::<BTreeMap<_, _>>(),
                             ),
-                            Value::from(metric.timestamp()),
-                        ));
+                        );
                     }
-                    _ => {
-                        // Unrecognized metric type
+
+                    // Set type and type related attributes
+                    if let (MetricValue::Counter { .. }, MetricKind::Incremental) =
+                        (data.value, data.kind)
+                    {
+                        if let Some(interval_ms) = data.time.interval_ms {
+                            metric_data.insert(
+                                "interval.ms".to_owned(),
+                                Value::from(interval_ms.get() as i64),
+                            );
+                        } else {
+                            // Incremental counter without an interval is worthless, skip this metric
+                            continue;
+                        }
+                        metric_data.insert("type".to_owned(), Value::from("count"));
+                    } else {
+                        // Anything that's not an incremental counter is considered a gauge, that is gauge and absolute counters metrics.
+                        metric_data.insert("type".to_owned(), Value::from("gauge"));
                     }
+
+                    metric_array.push(metric_data);
```

Comment:
It appears exactly one element is pushed per input element. Could this be a `map` then?